src/server.tsx

```diff
@@ -58,6 +58,14 @@ app.get('/healthcheck', (req: express.Request, res: express.Response) => {
 interface Epic {
     html: string;
     css: string;
+    js: string;
+}
+
+interface ComponentDescriptor {
+    name: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Component: React.FC<any>;
+    getInitScript: Function;
```

Comment:
Why not just `js`? I.e. what's the motivation to return a function here?

Reply:
We want this to be a function so it can check what JS code is needed for that component based on its props. In this particular case, we only want this function to return some JS code if `props.variant.showReminderFields` is `true`.

Reply:
But what about performing that logic in the component function? I.e. why lift it?

Reply:
The actual component function (the init code) is never called here. It's only called on the client so we need to check this prop logic on the server. I don't see where else would be best to check for this :/

Reply:
@andre1050 I've raised a branch to illustrate my thinking. Hopefully it at least clarifies my thoughts!